jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -6066,36 +6066,41 @@ public int indexOf(String subString, int start) {
 			char[] s1Value = s1.value;
 			char[] s2Value = s2.value;
 
-			if (enableCompression && (null == compressionFlag || (s1.count | s2.count) >= 0)) {
-				// both s1 and s2 are compressed
-				return helpers.intrinsicIndexOfStringLatin1(s1Value, s1len, s2Value, s2len, start);
-			} else if (s1.count < 0 && s2.count < 0) {
-				// both s1 and s2 are decompressed
-				return helpers.intrinsicIndexOfStringUTF16(s1Value, s1len, s2Value, s2len, start);
-			} else {
-				// s1 decompressed and s2 compressed
-				char firstChar = s2.charAtInternal(0, s2Value);
+			if (enableCompression) {
+				if (null == compressionFlag || (s1.count | s2.count) >= 0) {
+					// Both s1 and s2 are compressed
+					return helpers.intrinsicIndexOfStringLatin1(s1Value, s1len, s2Value, s2len, start);
+				} else if (s1.count < 0 && s2.count < 0) {
```

Comment:
Following the example above, this could be written as `(s1.count & s2.count) < 0`.

Reply:
Fixed.